Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/Utilities/ValidationUtil.cs

```diff
@@ -0,0 +1,305 @@
+            {
+                if (string.Equals(target, memberName))
+                {
+                    return true;
+                }
+
+                memberNameCount++;
+            }
+
+            return memberNameCount == 0 && string.IsNullOrEmpty(target);
+        }
+
+        /// <summary>
+        /// Finds an equivalent ValidationResult if one exists.
+        /// </summary>
+        /// <param name="collection">ValidationResults to search through.</param>
+        /// <param name="target">ValidationResult to find.</param>
+        /// <returns>Equal ValidationResult if found, null otherwise.</returns>
+        public static ValidationResult FindEqualValidationResult(this ICollection<ValidationResult> collection, ValidationResult target)
+        {
+            foreach (ValidationResult oldValidationResult in collection)
+            {
+                if (oldValidationResult.ErrorMessage == target.ErrorMessage)
+                {
+                    bool movedOld = true;
+                    bool movedTarget = true;
+                    IEnumerator<string> oldEnumerator = oldValidationResult.MemberNames.GetEnumerator();
+                    IEnumerator<string> targetEnumerator = target.MemberNames.GetEnumerator();
+                    while (movedOld && movedTarget)
+                    {
+                        movedOld = oldEnumerator.MoveNext();
+                        movedTarget = targetEnumerator.MoveNext();
+
+                        if (!movedOld && !movedTarget)
+                        {
+                            return oldValidationResult;
+                        }
+
+                        if (movedOld != movedTarget || oldEnumerator.Current != targetEnumerator.Current)
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Searches through all Bindings on the specified element and returns a list of BindingInfo objects
+        /// for each Binding that matches the specified criteria.
+        /// </summary>
+        /// <param name="element">FrameworkElement to search</param>
+        /// <param name="dataItem">Only return Bindings with a context element equal to this object</param>
+        /// <param name="twoWayOnly">If true, only returns TwoWay Bindings</param>
+        /// <param name="useBlockList">If true, ignores elements not typically used for input</param>
+        /// <param name="searchChildren">If true, searches through the children</param>
+        /// <param name="excludedTypes">The Binding search will skip all of these Types</param>
+        /// <returns>List of BindingInfo for every Binding found</returns>
+        public static List<BindingInfo> GetBindingInfo(this FrameworkElement element, object dataItem, bool twoWayOnly, bool useBlockList, bool searchChildren, params Type[] excludedTypes)
```

Comment:
Shouldn't this be added to Microsoft.Toolkit.UWP.UI?

Reply:
Same. We prefered to have a standalone DataGrid assembly.